urla134d66615_fikr4n_itl_java_tgz-pJ8-example_ExampleJ8/src/org/arabeyes/itl/hijri/UmmAlquraCalculator.java

```diff
@@ -0,0 +1,694 @@
+
+	   return(F);
+	}
+
+	/****************************************************************************/
+	/* Name:    GCalendarToJD                                                   */
+	/* Type:    Function                                                        */
+	/* Purpose: convert Gdate(year,month,day) to Julian Day                     */
+	/* Arguments:                                                               */
+	/* Input : Gregorian date: year:yy, month:mm, day:dd                        */
+	/* Output:  The Julian Day: JD                                              */
+	/****************************************************************************/
+	private static double GCalendarToJD(int yy, int mm, double dd)
+	{
+	   /* It does not take care of 1582 correction assumes correct
+	      calender from the past
+	    */
+	   int A, B, m, y;
+	   double T1, T2, Tr;
+
+	   if (mm > 2)
+	   {
+	      y = yy;
+	      m = mm;
+	   }
+	   else
+	   {
+	      y = yy - 1;
+	      m = mm + 12;
+	   }
+	   A = y / 100;
+	   B = 2 - A + A / 4;
+	   T1=ip (365.25 * (y + 4716));
+	   T2=ip (30.6001 * (m + 1));
+	   Tr=T1+ T2 + dd + B - 1524.5 ;
+
+	   return(Tr);
+	}
+
+	/****************************************************************************/
+	/* Name:    GLeapYear                                                       */
+	/* Type:    Function                                                        */
+	/* Purpose: Determines if  Gdate(year) is leap or not                       */
+	/* Arguments:                                                               */
+	/* Input : Gregorian date: year                                             */
+	/* Output:  0:year not leap   1:year is leap                                */
+	/****************************************************************************/
+	private static int GLeapYear(int year)
+	{
+	   int T = 0;
+
+	   if(year % 4 == 0)
+	      T = 1;	/* leap_year = 1; */
+
+	   if(year % 100 == 0)
+	   {
+	      /* years=100,200,300,500,... are not leap years */
+	      T=0;
+
+	      /* years=400,800,1200,1600,2000,2400 are leap years */
+	      if(year % 400 == 0) T = 1;
+	   }
+
+	   return(T);
+	}
+
+	/****************************************************************************/
+	/* Name:    GDateAjust                                                      */
+	/* Type:    Procedure                                                       */
+	/* Purpose: Adjust the G Dates by making sure that the month lengths        */
+	/*          are correct if not so take the extra days to next month or year */
+	/* Arguments:                                                               */
+	/* Input: Gregorian date: year:yg, month:mg, day:dg                         */
+	/* Output: corrected Gregorian date: year:yg, month:mg, day:dg              */
+	/****************************************************************************/
+	private static void GDateAjust(HijriCalculator.sDate mydate)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 485-528 into a separate method.